### ISSUE TITLE
Usage of "pattern_properties" with object

### ISSUE THREAD

**Issue by Stex** [parrish/json-schema_builder#7]
In JSON Schema, it is possible to define object properties based on regular expressions, even while still using fixed keys:

```json
{
  "type": "object",
  "properties": {
    "builtin": { "type": "number" }
  },
  "patternProperties": {
    "^S_": { "type": "string" },
    "^I_": { "type": "integer" }
  }
}
```

Looking into the [object](https://github.com/parrish/json-schema_builder/blob/master/lib/json/schema_builder/object.rb) class, I tried to translate this to your DSL as follows:

```ruby
class MySchema
  require 'json/schema_builder'
  include JSON::SchemaBuilder

  def schema
    object do
      pattern_properties do
        string '^S_'
        string '^I_'
      end

      number :builtin
    end
  end
end

puts JSON.pretty_generate(MySchema.new.schema.as_json)
```

This only leads to everything outside `pattern_properties` to be generated:

```json
{
  "type": "object",
  "properties": {
    "builtin": {
      "type": "number"
    }
  }
}
```

My next try was to specify the pattern properties as option for `object`, but this only lead to `patternProperties` without the actual patterns:

```ruby
object pattern_properties: [string('^S_')] do
   ...
end
```

```json
{
  "type": "object",
  "patternProperties": [
    {
      "type": "string"
    }
  ],
  "properties": {
    "builtin": {
      "type": "number"
    }
  }
}
```

---

Could you give me an example of how to correctly use pattern arguments in your DSL?
Thanks in advance!

**Comment by Haniyya** [comment]
I think @jirutka solved this problem [here](https://github.com/jirutka/json-schema_builder/commit/372d0694f68fe24abe267338ac19a14efa70f295). Don't think he made a PR for that though.

**Comment by Stex** [comment]
#3 indeed looks promising, especially with @parrish's comment regarding the block evaluation

**Comment by parrish** [comment]
Yep, #3 should fix this.  With respect to a `pattern_properties` block, it seemed easier to just build it based on Regexp named properties.

**Comment by Stex** [comment]
@parrish Could you release a new gem version now that this is merged?

**Comment by parrish** [comment]
:+1: Released in 0.7.1
